pandas/core/strings.py

```diff
@@ -1962,11 +1934,269 @@ def cons_row(x):
                 cons = self._orig._constructor
                 return cons(result, name=name, index=index)
 
-    @copy(str_cat)
-    def cat(self, others=None, sep=None, na_rep=None):
-        data = self._orig if self._is_categorical else self._data
-        result = str_cat(data, others=others, sep=sep, na_rep=na_rep)
-        return self._wrap_result(result, use_codes=(not self._is_categorical))
+    def _str_cat_los(self, input, ignore_index=False):
+        """
+        Auxiliary function for :meth:`str.cat`. Turn potentially mixed input
+        into list of Series (elements without an index must match the length of
+        the calling Series/Index).
+
+        Parameters
+        ----------
+        input : Series, DataFrame, np.ndarrary, list-like or list-like of those
+        ignore_index : Boolean
+            Determines whether to forcefully align with index of the caller
+
+        Returns
+        -------
+        tuple : first element: input transformed into list of Series
+            second element: Boolean whether FutureWarning should be raised
+        """
+
+        # once str.cat defaults to alignment, this function can be simplified;
+        # will not need `ignore_index` and the second boolean output anymore
+
+        from pandas.core.index import Index
+        from pandas.core.series import Series
+        from pandas.core.frame import DataFrame
+
+        # self._orig is either Series or Index
+        idx = self._orig if isinstance(self._orig, Index) else self._orig.index
+
+        if isinstance(input, Series):
+            los = [Series(input.values, index=idx) if ignore_index else input]
+            return (los, True)
+        elif isinstance(input, Index):
+            los = [Series(input.values,
+                          index=(idx if ignore_index else input))]
+            return (los, True)
+        elif isinstance(input, DataFrame):
```

Comment:
so you are adding 2-D ndarray and DF support, can you give me a compelling use-case for this.

Reply:
answered in the main thread